app/views/portal/show.rb

```diff
@@ -0,0 +1,871 @@
+        {
+          license_human: t('global.facet.reusability.restricted'),
+          license_name: t('global.facet.reusability.advanced-cc-by-nc-nd'),
+          license_CC_BY_NC_ND: true
+        }
+      elsif rights.index('http://creativecommons.org/licenses/by-nc-sa') == 0
+        {
+          license_human: t('global.facet.reusability.restricted'),
+          license_name: t('global.facet.reusability.advanced-cc-by-nc-sa'),
+          license_CC_BY_NC_SA: true
+        }
+      elsif rights.index('http://creativecommons.org/licenses/by-sa') == 0
+        {
+          license_human: t('global.facet.reusability.open'),
+          license_name: t('global.facet.reusability.advanced-cc-by-sa'),
+          license_CC_BY_SA: true
+        }
+      elsif rights.index('http://www.europeana.eu/rights/out-of-copyright-non-commercial') == 0
+        {
+          license_human: t('global.facet.reusability.restricted'),
+          license_name: t('global.facet.reusability.advanced-out-of-copyright-non-commercial'),
+          license_OOC: true
+        }
+      elsif rights.index('http://www.europeana.eu/rights/rr-f') == 0
+        {
+          license_human: t('global.facet.reusability.permission'),
+          license_name: t('global.facet.reusability.advanced-rrfa'),
+          license_RR_free: true
+        }
+      elsif rights.index('http://www.europeana.eu/rights/rr-p') == 0
+        {
+          license_human: t('global.facet.reusability.permission'),
+          license_name: t('global.facet.reusability.advanced-rrpa'),
+          license_RR_paid: true
+        }
+      elsif rights.index('http://www.europeana.eu/rights/rr-r/') == 0
+        {
+          license_human: t('global.facet.reusability.permission'),
+          license_name: t('global.facet.reusability.advanced-rrra'),
+          license_RR_restricted: true
+        }
+      elsif rights.index('http://creativecommons.org/publicdomain/mark') == 0
+        {
+          license_public: true,
+          license_name: t('global.facet.reusability.advanced-pdm'),
+          license_human: t('global.facet.reusability.open')
+        }
+      elsif rights.index('http://www.europeana.eu/rights/unknown') == 0
+        {
+          license_unknown: true,
+          license_name: t('global.facet.reusability.advanced-ucs'),
+          license_human: t('global.facet.reusability.permission')
+        }
+      elsif rights.index('http://www.europeana.eu/rights/test-orphan') == 0
+        {
+          license_orphan: true,
+          license_name: t('global.facet.reusability.advanced-orphan-work'),
+          license_human: t('global.facet.reusability.permission')
+        }
+      else
+        {
+          license_public: false,
+          license_name: 'unmatched rights: ' + rights
+        }
+      end
+    end
+
+    def creator_title
+      document.fetch('agents.prefLabel', []).first || render_document_show_field_value(document, 'dcCreator')
+    end
+
+    def media_items
```

Comment:
Cyclomatic complexity for media_items is too high. [23/6]<br>Assignment Branch Condition size for media_items is too high. [80.02/15]<br>Method has too many lines. [111/10]<br>Perceived complexity for media_items is too high. [27/7]